Clarify when running `challenge-install` is required

The alias `challenge-install` install the npm packages based on the content of `yarn.lock`. This command should be run each time `yarn.lock` changes when pulling a commit from the remote. This is an attempt to solve an "issue" that has been mentioned in a few discussions.

## Option 1

Do nothing. A developer may then bump into an error because the packages installed in `node_modules` are not sync with the content of `yarn.lock`, and thus not providing the dependencies that the code actually needs. Ideally here the developer will figure out that updating packages with solve the issue.

## Option 2

As the developer to manually run `challenge-install` each time that they pull code from remote, e.g. after pulling the latest version of `main` or the latest commit from a feature branch. The solution is to add this step to the documentation, which is easy. However, the developers may overlook this instructions and thus end up with the issue described in Option 1.

## Option 3

Programmatically detect when `yarn.lock` and `node_modules` are not in sync and run `challenge-install` or display a message to the user to ask to run `challenge-install`.

Here is an error I bumped into several times because of an out-of-sync `node_modules`.

```console
 FAIL   challenge-registry-ui  libs/challenge-registry/ui/src/lib/footer/footer.component.spec.ts
  ● Test suite failed to run
       
           Cannot find module '../build/Release/canvas.node'
           Require stack:
           - /workspaces/challenge-registry/node_modules/canvas/lib/bindings.js
           - /workspaces/challenge-registry/node_modules/canvas/lib/canvas.js
           - /workspaces/challenge-registry/node_modules/canvas/index.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/utils.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/MouseEvent-impl.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/generated/MouseEvent.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/living/interfaces.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/jsdom/browser/Window.js
           - /workspaces/challenge-registry/node_modules/jest-environment-jsdom/node_modules/jsdom/lib/api.js
           - ...
```

--- a/.husky/post-checkout
+++ b/.husky/post-checkout
@@ -13,5 +13,3 @@
 if [ "$3" = "1" ]; then
   yarn install --frozen-lockfile
 fi
-
-echo "The post-checkout hook ran. 🚀"

--- a/.husky/post-checkout
+++ b/.husky/post-checkout
@@ -1,0 +1,18 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+# From the post-checkout docs https://git-scm.com/docs/githooks#_post_checkout
+# The hook is given three parameters:
+#  - the ref of the previous HEAD
+#  - the ref of the new HEAD (which may or may not have changed)
+#  - a flag indicating whether the checkout was:
+#     - a branch checkout (changing branches, flag=1)
+#     - a file checkout (retrieving a file from the index, flag=0).
+
+# When the third script parameter is "1" we are executing a branch checkout
+if [ "$3" = "1" ]; then
+  # install npm packages
+  yarn install --frozen-lockfile
+fi
+
+echo "The post-checkout hook ran. 🚀"

--- a/.husky/post-checkout
+++ b/.husky/post-checkout
@@ -11,7 +11,6 @@
 
 # When the third script parameter is "1" we are executing a branch checkout
 if [ "$3" = "1" ]; then
-  # install npm packages
   yarn install --frozen-lockfile
 fi
 

--- a/.husky/post-merge
+++ b/.husky/post-merge
@@ -1,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+rm -rf node_modules && yarn install --frozen-lockfile

--- a/.husky/post-rewrite
+++ b/.husky/post-rewrite
@@ -1,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+yarn install --frozen-lockfile

--- a/package.json
+++ b/package.json
@@ -84,6 +84,7 @@
     "@typescript-eslint/eslint-plugin": "5.30.7",
     "@typescript-eslint/parser": "5.30.7",
     "babel-jest": "27.5.1",
+    "colors": "^1.4.0",
     "coveralls": "^3.1.1",
     "cypress": "^9.1.0",
     "eslint": "8.15.0",

